components/SpotifyInterface/SpotifyMakeRequest.c

```diff
@@ -32,222 +42,277 @@ bool Spotify_FindCode(char *Response, uint16_t SizeRes)
+    };
 
-/**
- * @brief This function sends a request to the Spotify API to retrieve the user's profile information.
- * @return This function does not return a value.
- */
-void Spotify_GetUserStatus(char *AccessToken)
-{
-    char Buf[LONG_BUF];
-    sprintf(Buf, "GET /v1/me/ HTTP/1.1\r\n"
-                 "Host: api.spotify.com\r\n"
-                 "Authorization: Bearer %s\r\n"
-                 "Content-Length: 0\r\n"
-                 "Connection: close\r\n\r\n",
-            AccessToken);
-    ESP_LOGI(TAG, "\n\nrequest for Spotify_GetUserStatus\n%s\nand size it =%d\n\n", Buf, strlen(Buf));
-    char URL[SMALL_BUF] = "https://api.spotify.com";
-    char Server[SMALL_BUF] = "api.spotify.com";
-    HttpsHandler(Buf, sizeof(Buf), URL, sizeof(URL), Server, sizeof(Server));
-}
+    // Initialize HTTP client with custom configuration
+    esp_http_client_handle_t client = esp_http_client_init(&custom_config);             // apply configuration to the client object
 
-/**
- * @brief This function sends a request to the Spotify API to retrieve the user's top item.
- * @return This function does not return a value.
- */
-void Spotify_GetUserTopItems(char *AccessToken)
-{
-    char Buf[LONG_BUF];
-    sprintf(Buf, "GET /v1/me/top/artists HTTP/1.1\r\n"
-                 "Host: api.spotify.com\r\n"
-                 "Authorization: Bearer %s\r\n"
-                 "Content-Length: 0\r\n"
-                 "Connection: close\r\n\r\n",
-            AccessToken);
-    ESP_LOGI(TAG, "\n\nrequest for Spotify_GetUserTopItems\n%s\nand size it =%d\n\n", Buf, strlen(Buf));
-    char URL[SMALL_BUF] = "https://api.spotify.com";
-    char Server[SMALL_BUF] = "api.spotify.com";
-    HttpsHandler(Buf, sizeof(Buf), URL, sizeof(URL), Server, sizeof(Server));
-}
+    if (client == NULL) {
+        ESP_LOGE(TAG, "Failed to create the HTTP client");
+        return;
+    }
 
-/**
- * @brief This function sends a request to the Spotify API to retrieve the profile information of a specific user.
- * @param[in] UserId_t The ID of the user whose profile information is to be retrieved.
- * @return This function does not return a value.
- */
-void Spotify_GetUserProfile(char *UserId_t, char *Token)
-{
-    char Buf[LONG_BUF];
-    char UserId[50];
-    strcpy(UserId, UserId_t);
-    sprintf(Buf, "GET /v1/users/%s HTTP/1.1\r\n"
-                 "Host: api.spotify.com\r\n"
-                 "Authorization: Bearer %s\r\n"
-                 "Content-Length: 0\r\n"
-                 "Connection: close\r\n\r\n",
-            UserId, Token);
-    ESP_LOGI(TAG, "\n\nrequest for Spotify_GetUserProfile\n%s\nand size it =%d\n\n", Buf, strlen(Buf));
-    char URL[SMALL_BUF] = "https://api.spotify.com";
-    char Server[SMALL_BUF] = "api.spotify.com";
-    HttpsHandler(Buf, sizeof(Buf), URL, sizeof(URL), Server, sizeof(Server));
+    // Set headers for authentication
+    char authorizationHeader[MEDIUM_BUF];
```

Comment:
please use "Pascal" format for naming .

Reply:
for local variables we use camelCase. change applied